### DISCUSSION TITLE
Debug build onnxruntime_test_all timeout

### DISCUSSION THREAD

**Discussion by hahne** [microsoft/onnxruntime#17632] [Other Q&A] [answered]
Hello,

I can build the release version of ONNX runtime and the test run through without any issues.
I can also manually run the debug version of onnxruntime_test_all.exe and it runs through without any errors.

But when I build ONNX runtime with tests in debug mode, I get the following error:

> 8/8 Test #8: onnxruntime_logging_apis_test ...........   Passed    0.15 sec
> 
> 88% tests passed, 1 tests failed out of 8
> 
> Total Test time (real) = 1541.98 sec
> 
> The following tests FAILED:
>           1 - onnxruntime_test_all (Timeout)
> Errors while running CTest

I tried to use the command line argument `--test_all_timeout` but it seems to be ignored, as I get:

> Test timeout computed to be: 1500

Any idea what the problem is here? Or is there another way to increase the timeout?

Thanks,
Johannes

**Comment by snnn** [comment, marked as answer]
See the change I made in PR #16856 on file cmake/onnxruntime_unittests.cmake

**Reply by hahne** [reply]
Nice, that worked!
